packages/meditrak-server/src/routes/GETUserAccounts/assertUserAccountPermissions.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { keyBy } from 'lodash';
+import {
+  BES_ADMIN_PERMISSION_GROUP,
+  TUPAIA_ADMIN_PANEL_PERMISSION_GROUP,
+} from '../../permissions/constants';
+
+export const hasUserAccountPermissions = async (accessPolicy, models, userAccount) => {
+  const entityPermissions = await models.userEntityPermission.find({ user_id: userAccount.id });
+  const entities = await models.entity.findManyById(entityPermissions.map(ep => ep.entity_id));
+  const countryCodes = entities.map(e => e.country_code).filter(c => c !== 'DL');
+  if (countryCodes.length === 0) {
+    // User only has access to Demo Land, and it got filtered out
+    return true;
+  }
+
+  return accessPolicy.allowsAll(countryCodes, TUPAIA_ADMIN_PANEL_PERMISSION_GROUP);
+};
+
+export const assertUserAccountPermissions = async (accessPolicy, models, userAccount) => {
+  const hasPermission = await hasUserAccountPermissions(accessPolicy, models, userAccount);
+  if (!hasPermission) {
+    throw new Error('Need Admin Panel access to all countries this user has access to');
```

Comment:
`assertAnyPermissions` will actually combine error messages from the individual checks. The resultant error looks something like:

```One of the following needs to be satisfied:
Need BES Admin access
Need Admin Panel access to all countries this user has access to```

Because of that I left the error message specific to the case this check was implemented for. I think it also makes the error message more accurate.

Reply:
ah right sorry I was a bit confused 😄